base/loading.jl

```diff
@@ -3131,7 +3131,7 @@ function replace_depot_path(path::AbstractString)
             depot = dirname(depot)
         end
 
-        if startswith(path, depot)
+        if startswith(path, string(depot, Filesystem.pathsep())) || path == depot
```

Comment:
I think you want to do something like https://github.com/JuliaLang/julia/blob/85f4db2eaa15fedf738f8a8c16c7168024780f7c/base/file.jl#L235

Reply:
It looks like I keep making the same suggestions: https://github.com/JuliaLang/julia/pull/51892#discussion_r1373628965

Reply:
I think that would falsely insert a `@depot` when `path="/depot/julia-rc/somefile.jl"` and `depot="/path/julia"`, which is what this PR fixes.

Reply:
But thanks for making me double check, I realized I had the paths wrong in the tests.